File: migration-get-data.py
import requests
import json
import argparse
import pprint

class SalsahError(Exception):
    """Error in getting data from SALSAH"""
    pass

class Salsah:
    """
    Class to gather data from SALSAH
    """
    def __init__(self, server, user, password):
        """
        Constructor requireing the server adress, the user and password of SALSAH
        :param server: Adress of the serer, e.g http://data.dasch.swiss
        :param user: Username for SALSAH
        :param password: The password
        """
        self.server = server
        self.user = user
        self.password = password

    def get_all_obj_ids(self, project, start_at=0, show_nrows=-1):
        """
        Gets all objects ID's of a given project
        :param project: SALSAH Name of the project
        :return: List of all object (resource) ID's
        """
        payload = {
            'searchtype' : 'extended',
            'filter_by_project' : project
        }

        if nrows > 0:
            payload['show_nrows'] = show_nrows
            payload['start_at'] = start_at

        req = requests.get(self.server + '/api/search/', params=payload, auth=(self.user, self.password))
        result = req.json()
        if result['status'] != 0:
            raise SalsahError("SALSAH-ERROR:\n" + result['errormsg'])

        else:
            obj_ids = list(map(lambda a: a['obj_id'], result['subjects']))

            return obj_ids

    def get_resource(self, res_id):
        req = requests.get(self.server + '/api/resources/' + res_id, auth=(self.user, self.password))
        result = req.json()
        if result['status'] != 0:
            raise SalsahError("SALSAH-ERROR:\n" + result['errormsg'])

        else:
            return result


parser = argparse.ArgumentParser()
parser.add_argument("server", help="URL of the SALSAH server")
parser.add_argument("-u", "--user", help="Username for SALSAH")
parser.add_argument("-p", "--password", help="The password for login")
parser.add_argument("-n", "--nrows", type=int, help="number of records to get, -1 to get all")
parser.add_argument("-s", "--start", type=int, help="Start at record with given number")
args = parser.parse_args()

user = 'root' if args.user is None else args.user
password = 'SieuPfa15' if args.password is None else args.password
start = args.start;
nrows = -1 if args.nrows is None else args.nrows

con = Salsah(args.server, user, password)
res_ids = con.get_all_obj_ids('postcards', start, nrows)
resources = list(map(con.get_resource, res_ids))

pprint.pprint(resources)

File: migration-get-data.php
<?php
/**
 * Created by PhpStorm.
 * User: rosenth
 * Date: 05.09.18
 * Time: 16:57
 */

$GLOBALS['username'] = 'root';
$GLOBALS['password'] = 'SieuPfa15';



function get_all_res_ids($project, $show_nrows = -1, $start_at = 0) {

    $query = array(
        'searchtype' => 'extended',
        'filter_by_project' => $project
    );

    if ($show_nrows > 0) {
        $query['show_nrows'] = $show_nrows;
    }
    if ($start_at > 0) {
        $query['start_at'] = $start_at;
    }
    $cid = curl_init('http://data.dasch.swiss/api/search?' . http_build_query($query));
    curl_setopt($cid, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($cid, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
    curl_setopt($cid, CURLOPT_KEYPASSWD, $GLOBALS['username'].':'.$GLOBALS['password']);
    if (($jsonstr = curl_exec($cid)) === false) {
        die('curl_exec failed for property "'.$restype.'"!'.PHP_EOL);
    }
    curl_close($cid);
    $retdata = json_decode($jsonstr);
    if ($retdata->status != 0) {
        print_r(retdata);
        die('curl_exec failed for project  "'. $project .'"!'.PHP_EOL);
    }

    //print_r($retdata); die();

    $res_ids = array_map(function($ele) {
        return $ele->obj_id;
    }, $retdata->subjects);

    return $res_ids;

}
//=============================================================================


$project_name = NULL;
$outfile = NULL;

function print_usage() {
    echo 'usage: ', $_SERVER['argv'][0], ' -project project-name xml-dump-file', PHP_EOL, PHP_EOL;
    echo '  -project project-name: dump data from given project.', PHP_EOL;
    echo PHP_EOL;
}
//=============================================================================


for ($i = 1; $i < $_SERVER['argc']; $i++) {
    switch ($_SERVER['argv'][$i]) {
        case '-project': {
            $i++;
            $project_name = $_SERVER['argv'][$i];
            break;
        }
        case '-list': {
            break;
        }
        default: {
            $outfile = $_SERVER['argv'][$i];
        }
    }
}

if (is_null($project_name) || is_null($outfile)) {
    print_usage();
    die();
}

$res_ids = get_all_res_ids($project_name, 10, 0);

print_r ($res_ids);